keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/KDEVertex.java

```diff
@@ -45,8 +45,8 @@ public KDEVertex(List<Double> samples, double bandwidth) {
     }
 
     private DoubleTensor getDiffs(DoubleTensor x) {
-        DoubleTensor diffs = DoubleTensor.zeros(new long[]{samples.getShape()[0], x.getShape()[1]});
-        return diffs.plusInPlace(x).minusInPlace(samples).divInPlace(bandwidth);
+        DoubleTensor diffs = DoubleTensor.zeros(samples.getShape()[0], x.getShape()[0]);
+        return diffs.plusInPlace(x.reshape(1, x.getShape()[0])).minusInPlace(samples.reshape(samples.getShape()[0], 1)).divInPlace(bandwidth);
```

Comment:
This looks really gnarly. Can we pull out these vector shapes into variables? `1, x.getShape()[0]` etc. Or introduce a concept of a `reshapeColumnVector` or `reshapeRowVector` that just takes a length?

Reply:
It shouldn't be done very often. Those helpers might be useful but we should add them as another task.